pkg/packetbrokeragent/agent.go

```diff
@@ -310,20 +311,28 @@ func New(c *component.Component, conf *Config, opts ...Option) (*Agent, error) {
 
 	newTaskConfig := func(id string, fn component.TaskFunc) *component.TaskConfig {
 		return &component.TaskConfig{
-			Context: c.Context(),
+			Context: ctx,
 			ID:      id,
 			Func:    fn,
 			Restart: component.TaskRestartOnFailure,
 			Backoff: component.DialTaskBackoffConfig,
 		}
 	}
 	if a.forwarderConfig.Enable && a.dataPlaneAddress != "" {
-		c.RegisterTask(newTaskConfig("pb_publish_uplink", a.publishUplink))
-		c.RegisterTask(newTaskConfig("pb_subscribe_downlink", a.subscribeDownlink))
+		for i := 0; i < publishTaskCount; i++ {
+			c.RegisterTask(newTaskConfig("pb_publish_uplink", a.publishUplink))
+		}
+		for i := 0; i < subscribeTaskCount; i++ {
+			c.RegisterTask(newTaskConfig("pb_subscribe_downlink", a.subscribeDownlink))
+		}
```

Comment:
So now we will have multiple connections; one per subscription and per publisher, right? With standard config, that's already 10 connections.

We don't really need this now, also because the gRPC client internals may use multiple TCP connections. What we do need is multiple streams for subscriptions, but the connection may be the same.